### ISSUE TITLE
[스터디] 딥러닝의 기본

### ISSUE THREAD

**Issue by unizard** [kangdain/youtuboard#4]
### 시즌 1 - 딥러닝의 기본 부터 시작합시다.
https://hunkim.github.io/ml/

진행사항은 본 이슈 코멘트로 달아주세요.
강의가 길지 않으니 들을만 할꺼예요. 하루에 비디오 강의 3개정도면 좋을 듯해요.
보다가 모르는 부분은 언제든지 코멘트로 질문 하세요.

----

## 온라인 강의 및 실습 진행 상황
- [X] Lecture0: 수업의 개요와 일정
- [X] Lecture1: Tensorflow 설치 및 기본적인 Operations
- [X] Lecture2: Linear Regression의 Hypothesis 와 Cost 설명 
- [x] Lecture2-Lab: Tensorflow 실습
- [X] Lecture3: Linear Regression의 Cost 최소화 알고리즘 원리 설명 
- [x] Lecture3-Lab: Tensorflow 실습
- [X] Lecture4: Multi-variable linear regression 
- [x] Lecture4-Lab: Tensorflow 실습
- [X] Lecture5: Logistic Classification의 가설함수 정의 / Cost 합수 설명 
- [x] Lecture5-Lab: Tensorflow 실습
- [X] Lecture6: Softmax Regression 기본 개념 소개 / Softmax Classifier의 Cost 함수
- [x] Lecture6-Lab: Tensorflow 실습
- [X] Lecture7: 학습 rate, Overfitting, 그리고 일반화
- [x] Lecture7-Lab: Tensorflow 실습 / MNist Dataset
- [X] Lecture8: 딥러닝의 기본 개념: 시작과 XOR 문제 / Back-propagation
- [x] Lecture8-Lab: Tensor Manipulation
- [x] Lecture9: XOR 문제 딥러닝으로 풀기 / 10분안에 미분 정리하기 / 딥넷트워 학습 시키기
- [x] Lecture9-Lab: Neural Net for XOR / Tensorboard
- [x] Lecture10: Sigmoid / ReLU / Weight 초기화 / 넷트워 모듈 다루기
- [x] Lecture10-Lab: 실습
- [x] Lecture11: CovNet / Max Pooling / Full Network / 응용
- [x] Lecture11-Lab: TF CNN / MNIST / CNN Class, Layers, Ensemble 
- [x] Lecture12: RNN 
- [x] Lecture12-Lab: Stacked RNN + Softmax Layer / Dynamic RNN 
- [ ] Lecture13-Lab: AWS에서 GPU돌려보기
- [ ] Lecture14-Lab: AWS에서 저렴하게 Spot Instance 사용하기

**Comment by kangdain** [comment]
네 알겠습니다 :-)

**Comment by kangdain** [comment]
Lecture 00 -Lecture03

**Comment by kangdain** [comment]
Lecture07-1 ~ Lecture 08-1

**Comment by unizard** [comment]
@kangdain 금일 강의는 이런걸 배웠다. 한줄요약 부탁드려요. ^^

**Comment by unizard** [comment]
@kangdain 스터디 진행상황 정리를 잘 부탁드립니다.  :)

**Comment by kangdain** [comment]
@unizard  Lab2 ~ Lab4: Tensoflow 를 이용한 Linear Regression실습 -  cost 최소화, Multi-variable linear regression

**Comment by kangdain** [comment]
@unizard Lab5 ~ Lab7 : Tensorflow 실습 - Logistic classification , softmax classification

**Comment by kangdain** [comment]
@unizard 
- Lab08,Lec9-1  : Tensormanipulation :  Shape ,Rank , Axis 예제 및 shape, rank 다르더라도 tensor 연산이 가능한 broadcasting과 그외 많이 사용되는 reshape, one hot, reduce_mean 등의 연산을 알 수 있었다. (월)
- Lex9-x ,Lec9-2: 미분의 기본 개념 및 실제값과 예측값 사이에서 나타난 오류(= error, cost ) 를 통해 미분값을 계산하는 Backpropagation이 실제로 어떻게 동작하는 지에 대해 이해하기위하여 chain rule을 배웠다.

**Comment by kangdain** [comment]
@unizard  Lab9-1,Lab9-2

xor을 위한 딥러닝 / TensorBoard 사용법 
 
1. log할 tensor 를 정한다.
( 레이어별 보다 정돈된 그래프 계층을 보려면  scope 를 이용)
2. merge_all 을 통해 사용할 tensor들을 합친다. 
3. session을 열어 어느 디렉토리에을 용 할 것이지를 정한다.
4. session 을 통해 실행 시킨 값(summary)을 add_summary에 넣어 실행
5. command line 에서 tensor board  실행

**Comment by kangdain** [comment]
@unizard  Lec 10-1 ~ 10-3
lec10-1 :NN 에서 Sigmoid 사용시 1보다 작을 수를 곱해 나아가기 때문에 chain rule 적용시 경사도가 사라져 학습하기 어렵다는 문제가 발생한다. 때문에 0보다 작을 경우 동작하지 않고, 0보다 클 경우에만 값이 비례하여 커지는 ReLU를 사용하여 이와 같은 문제를 해결 할 수 있다. 
lec10-2: weight 초기값 설정의 중요성 - 같은 코드이더라도 weight 값을 어떻게 설정 하는냐에 따라 성능 차이가 크기 때문에 weight를 설정 시 여러 가지 경우의 수를 고려하여 다양한 시도를 해볼 수 있다. (but , 모든 값을 0 으로 할 경우 학습불가.)
lec10-3: Dropout & 앙상블 
NN에서 복잡한 노드들을 간단하게 표현 하기 위하여 랜덤으로 특정 뉴런들을 kill 하는 Dropout을 사용하여 성능을 향상 시킬수 있다. => Dropout
똑같은 형태의 NN을 만들어 학습을 시킨뒤, 마지막에 합쳐서 결과를 낸다.  => 앙상블

**Comment by kangdain** [comment]
Lec11-2,Lec11-3,Lab11-1 : CNN 에서 convolution layer를 sampling 하여 새로운 레이어를 얻는 pooling 과 처음부터 끝까지 하나로 연결되어있는 layer(Fully connected layer ) 개념 및 실습을 통해 CNN 이 이미지를 판단하는 과정에 대해 알수 있었다.

**Comment by kangdain** [comment]
Lab11-2,Lab11-3,Lec12 : 음성인식, 자연어처리 등의 sequence data 를 처리하는 모델인 RNN 은  재귀를 통해 단계(step)을  이전 가중치를 기억하고 이를 학습에 반영하여 다음 값을  예측(y = Wx + b )할 수 있다.

**Comment by kangdain** [comment]
Lab12-1,12-2,12-3 : 
dynamic_rnn 에서 cell을 생성시키는 부분과 cell을 가지고 학습시키는 부분을 나누어 원하는 형태의 cell  로 바꾸어 사용할 수 있다. 
RNN은 중간에 레이어를 많이 사용 할 수록 학습을 하는데 어려움이 있다. 이를 해결하기위한 방법으로 LSTM, GRU와 같은 모델이 많이 사용된다.

**Comment by kangdain** [comment]
Lab 12-4,12-5,12-6 : RNN 을 이용하여 시간에 따라서 값이 변하는 타임 시리즈 데이터를(ex. 주식시장데이터 예측) 예측할 수 있다.
